xbmc/filesystem/FileCache.cpp

```diff
@@ -235,7 +292,7 @@ void CFileCache::Process()
   }
 
   // create our read buffer
-  std::unique_ptr<char[]> buffer(new char[m_chunkSize]);
+  auto buffer = std::make_unique<std::vector<char>>(m_readSize);
```

Comment:
It seems that the natural way to handle a buffer with std::vector should be:

In FileCache.h
`std::vector<char> m_buffer;`

Here just:
`m_buffer.resize(m_readSize);`

Reply:
Since the thread is persistent for the life of the `CFileCache` instance I'm not sure it'd make a whole lot of difference, either way we're creating exactly one vector (but in your example potentially allocating the internal buffer twice, at creation and then at resize).

Style-wise I personally prefer to keep it local since it's a smaller scope.

Reply:
> I personally prefer to keep it local since it's a smaller scope.

Then maybe:
`std::vector<char> buffer(m_chunkSize);`

Reply:
Yes, that makes sense!